sync-core/src/main/java/com/cloudant/http/HttpConnectionFilter.java

```diff
@@ -0,0 +1,37 @@
+package com.cloudant.http;
+
+/**
+ * Created by tomblench on 30/03/15.
+ */
+
+/**
+ A <code>HttpConnectionFilter</code> can either be configured as a Request Filter or a Response Filter:
+
+ A Request Filter is run before the request is made to the server.
+
+ A Response Filter is run after the response is obtained from the
+ server but before the output stream is returned to the original client. The Response
+ Filter enables two main behaviours:
+
+ - Modifying the response for every request
```

Comment:
What modifications are "safe" here? Can the response filter read the response content stream for example, can/how would it modify that?